README.md

```diff
@@ -1,34 +1,141 @@
 
-### Requirement Notes
-* You may not use the predefined C++ Standard Template Library (STL) classes.
-* This lab must run on the Linux machines. The pass-off scripts will work only on those machines
-* For this lab only, due to the nature of template classes, you are strongly encouraged to implement the entire class in only a '.h'.  All classes that are not templates must be implemented by a '.h' and a '.cpp'
-* Make sure that you pass all 5 tests before attempting to fix any memory leaks.  If anything causes the program to crash (or if the pass-off script is unable to finish testing your code) the pass-off script may terminate before it is able to delete your LinkedList, which will always result in memory leaks.
+# Testing your code
+
+To test your code, you have two options:
+
+1. Write your own test code in `scratch.cpp`
+2. Run the tests contained in `tests.cpp`. These are the same tests that the
+   autograder will run. To choose a part to test, you need to pass the part's
+   number as a command-line argument. To do that, you'll need to open the
+   drop-down menu to the left of the green play button, open the drop-down menu
+   under "tests", click "Edit", and put the part number in the "Program
+   arguments" field. The valid options are 1, 2, 3, 4, 5, 6, and all. In
+   `tests.cpp`, there are six functions, one to test each part. They are called
+   `part1`, `part2`, etc. You can jump to the appropriate function to see the
+   code that the autograder will run on your linked list. The autograder is
+   expecting the output of `tests.cpp` to exactly match `part1_key.txt` for part
+   1, `part2_key.txt` for part 2, etc.
+
+To switch between the two methods of testing listed above, you can use the
+drop-down menu to the left of the green play button. You'll click on either
+"scratch" or "tests", depending on which one you want to use.
+
+# Implementing the lab
+
+If you find yourself confused, you might find
+[this video](https://youtu.be/zRdZaBSqjEM) helpful. Otherwise, you can work
+through the tasks below to implement your linked list class:
+
+## Part 1 (20 points) - Node class, constructor, `toString()`, and `insertHead()`
+
+Let's start off by creating a constructor and implementing `insertHead` and
+`toString` so that we can see what's going on in our linked list.
+
+Before implementing any methods, you'll need to create a `Node` class that can
+hold a value and point to the next node. After that, you'll need to add a
+private data member to the `LinkedList` for your `head` pointer. Once you've
+done that, you're ready to start implementing `LinkedList`'s methods.
+
+The constructor should be pretty simple; all it really needs to do is set `head`
+to `nullptr`.
+
+After getting the constructor working, implement the `insertHead` method. This
+method should:
+1. Check that the given value is not already in the list. If it is, `insertHead`
+   should return without adding anything to the list.
```

Comment:
Why? That is the interface for a set, not a list.

If we want to test that behavior, let's add a `bool contains(T item);` to the interface.